src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -109,46 +123,63 @@ public static CollectionSettings getCollectionSettings(
       boolean vectorEnabled,
       int vectorSize,
       SimilarityFunction similarityFunction,
-      String vectorize,
+      String comment,
       ObjectMapper objectMapper) {
-    // parse vectorize to get vectorizeServiceName and modelName
-    if (vectorize != null && !vectorize.isBlank()) {
-      return createCollectionSettingsFromJson(
-          collectionName, vectorEnabled, vectorSize, similarityFunction, vectorize, objectMapper);
-    } else {
-      return new CollectionSettings(
-          collectionName, vectorEnabled, vectorSize, similarityFunction, null, null);
-    }
+    return createCollectionSettings(
+        collectionName, vectorEnabled, vectorSize, similarityFunction, comment, objectMapper);
   }
 
-  private static CollectionSettings createCollectionSettingsFromJson(
+  private static CollectionSettings createCollectionSettings(
       String collectionName,
       boolean vectorEnabled,
       int vectorSize,
       SimilarityFunction function,
-      String vectorize,
+      String comment,
       ObjectMapper objectMapper) {
-    try {
-      JsonNode vectorizeConfig = objectMapper.readTree(vectorize);
-      String vectorizeServiceName = vectorizeConfig.path("service").textValue();
-      JsonNode optionsNode = vectorizeConfig.path("options");
-      String modelName = optionsNode.path("modelName").textValue();
-      if (vectorizeServiceName != null
-          && !vectorizeServiceName.isEmpty()
-          && modelName != null
-          && !modelName.isEmpty()) {
+
+    if (comment == null || comment.isBlank()) {
+      return new CollectionSettings(
+          collectionName, vectorEnabled, vectorSize, function, null, null, null);
+    } else {
+      try {
+        String vectorizeServiceName = null;
+        String modelName = null;
+
+        JsonNode commentConfig = objectMapper.readTree(comment);
```

Comment:
Fine to leave catch block further down; but alternate that I sometimes I use is to do

```
final JsonNode commentConfig;

try {
  commentConfig = objectMapper.readTree(comment);
} catch (JsonProcessingException e) {
  throw new RuntimeExcepion(e);
}
```

to make it clear where theoretical exception comes from.

This of course assumes there aren't multiple sources.